### PR TITLE
S04/products list

### PR DESCRIPTION
**Pull request by NikitaIASA**
**S4 - Products page - products list**

- UI 
- Configure JSON that you need
- Add products to mock API
- Call API and show products

The UI and logic for getting products was developed. Categories and brands are displayed dynamically in the header. Components were also made to show errors and skeletons at the time of loading.

### CODE REVIEW

File: src/components/Products/ProductList/ProductList.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  const skeletons = [...new Array(5)].map((_, index) => (
+    <ProductSkeleton key={index} />
+  ));
+
+  if (status === STATUSES.LOADING) {
+    return <div className="product-list">{skeletons}</div>;
+  }
+
+  return (
+    <div className="product-list">
```
Comment (Yuliia143) [inline src/components/Products/ProductList/ProductList.tsx]: Why product-list, but div ?

File: src/components/Products/ProductCard/Stars/Stars.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import emptyStar from "@assets/images/empty-heart.svg";
+import star from "@assets/images/star.svg";
+
+interface StarProps {
+  rating: number;
+}
+
+export const Stars: FC<StarProps> = ({ rating }) => {
+  return (
+    <div className="product-card__stars">
```
Comment (Yuliia143) [inline src/components/Products/ProductCard/Stars/Stars.tsx]: Shouldn't this be a list?

File: src/components/Products/ProductCard/Stars/Stars.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      {[...new Array(5)].map((_, index) => (
```
Comment (Yuliia143) [inline src/components/Products/ProductCard/Stars/Stars.tsx]: Magic number

File: src/components/Products/ProductCard/ProductCard.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+          <Link to={`${ROUTE_PATHS.PRODUCTS}/${title}`}>
+            <h4 className="product-card__title">{title}</h4>
+          </Link>
+          <p className="product-card__description">{description}</p>
+          <Stars rating={rating} />
+          <ul className="product-card__details">
+            {Object.entries(details).map(([key, value]) => (
+              <li className="product-card__details-item" key={key}>
+                <p className="product-card__details-title">
+                  {key.charAt(0).toUpperCase() + key.slice(1)}
```
Comment (Yuliia143) [inline src/components/Products/ProductCard/ProductCard.tsx]: Logic in JSX

File: src/components/Products/ProductList/ProductList.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+import "./ProductList.scss";
+
+interface ProductListProps {
+  items: IProduct[];
+  status: string;
+}
+
+export const ProductList: FC<ProductListProps> = ({ items, status }) => {
+  const skeletons = [...new Array(5)].map((_, index) => (
```
Comment (Yuliia143) [inline src/components/Products/ProductList/ProductList.tsx]: Magic number

File: src/types/products.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    answer: string;
+}
+
+export interface IExtraInfo {
+    descriptions: {
+        origins?: string;
+        howToCook?: string;
+        features?: string;
+        warranty?: string;
+    };
```
Comment (Yuliia143) [inline src/types/products.ts]: Think it's better to have separate file for each type in this case

Comment (NikitaIASA) [reply src/types/products.ts]: Done.

File: src/components/Products/ProductList/ProductList.tsx (hunk 5 above)
Comment (NikitaIASA) [reply src/components/Products/ProductList/ProductList.tsx]: Done.

File: src/components/Products/ProductCard/ProductCard.tsx (hunk 4 above)
Comment (NikitaIASA) [reply src/components/Products/ProductCard/ProductCard.tsx]: Done.

File: src/components/Products/ProductCard/Stars/Stars.tsx (hunk 3 above)
Comment (NikitaIASA) [reply src/components/Products/ProductCard/Stars/Stars.tsx]: Done.

File: src/components/Products/ProductCard/Stars/Stars.tsx (hunk 2 above)
Comment (NikitaIASA) [reply src/components/Products/ProductCard/Stars/Stars.tsx]: Done.

File: src/components/Products/ProductList/ProductList.tsx (hunk 1 above)
Comment (NikitaIASA) [reply src/components/Products/ProductList/ProductList.tsx]: Done.

File: src/components/Products/ProductCard/ProductCard.tsx [hunk 7]
```diff
@@ -14,11 +14,28 @@ interface ProductListProps {
...
+    model,
+    delivery,
+    stock,
+    price,
+    shipping,
+  } = product;
+
+  const details = {
+    Brand: brand,
+    Modle: model,
```
Comment (Yuliia143) [inline src/components/Products/ProductCard/ProductCard.tsx]: Modle?

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**